### PR TITLE
Add sound mode support

### PR DESCRIPTION
**Pull request by starkillerOG**
**Description:**
- Two aditional configuration options: sound_mode and sound_mode_dict
- Aditional service: media_player/select_sound_mode

Front end pull request: home-assistant/home-assistant-polymer#815

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#13706

## Checklist:

  - [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`. 
  - [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/media_player.denonavr.markdown
Comment (frenck) [inline source/_components/media_player.denonavr.markdown]: We are trying to keep the examples as minimal as possible. Please remove the added optional configuration paramters.

File: source/_components/media_player.denonavr.markdown
Comment (frenck) [inline source/_components/media_player.denonavr.markdown]: - `Dictonary` -> `Dictionary`

File: source/_components/media_player.denonavr.markdown
Comment (frenck) [inline source/_components/media_player.denonavr.markdown]: - `diffrent` -> `different`
- Backquote sound_mode_dict keyword

**Review by frenck — CHANGES_REQUESTED** [review]
@starkillerOG I've reviews the docs and left some comments. Could you please take a look? Thx already! 👍

File: source/_components/media_player.denonavr.markdown
Comment (starkillerOG) [reply source/_components/media_player.denonavr.markdown]: How do you make a backquote in this "code"?

File: source/_components/media_player.denonavr.markdown
Comment (starkillerOG) [reply source/_components/media_player.denonavr.markdown]: Thanks for checking my bad spelling

File: source/_components/media_player.denonavr.markdown
Comment (starkillerOG) [reply source/_components/media_player.denonavr.markdown]: I removed the sound_mode. But I would like to keep the sound_mode_dict in the example because this is a somewhat difficult parameter, for example if you put " " around it and make it a string it will not work anymore. And if I remember corectly the values in this dict schould be uppercase, otherwise the values received from the AVR will not match. And a dictionary is in general a little more difficult then just a string or a number for the basic users.

Furthermore the hole zones parameter is also in the example and you do not need that in general and is also optional if you don't use second zones on your AVR.

**Comment by starkillerOG** [comment]
@frenck Thanks for your review, I made changes based on your comments.
Could you maybe also look at the backend and frontend pull request for this sound mode support?
Front end: home-assistant/home-assistant-polymer#815
Back end: home-assistant/home-assistant#11764

**Comment by frenck** [comment]
Thanks for the adjustments @starkillerOG 🎖 
This PR can be merged as soon as both parent PRs have been merged.

I'm sorry but I do not review those type of PRs.

**Comment by starkillerOG** [comment]
@frenck could you check your request for changes as addressed, so the red cross will go away and it can be merged once the other 2 PR's are accepted?

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
@starkillerOG Sorry simply forgot to hit the approve button, my bad... 🤦‍♂️

**Comment by starkillerOG** [comment]
Still waiting until my PR of the denonavr library is merged.

**Review by frenck — CHANGES_REQUESTED** [review]
@starkillerOG Could you please take a look at the merge conflicts?

**Comment by starkillerOG** [comment]
@frenck I think I resolved the conflict, but I am still waiting until the backend code can be merged.

**Review by frenck — DISMISSED** [review]

**Comment by starkillerOG** [comment]
@frenck I think I might have undone your changes, sorry I just wanted to correct the syntax.
Can you do the changes again?
Sorry.

**Comment by starkillerOG** [comment]
Thank you @frenck

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
No! Thank you 👍

**Comment by frenck** [comment]
Closing this PR, since the parent PR has been closed as well.

**Pull request closed without merging**
